code/controllers/subsystems/configuration.dm

```diff
@@ -0,0 +1,151 @@
+
+	for(var/filename in config_lines)
+		if(fexists(filename))
+			fdel(filename)
+		var/write_file = file(filename)
+		to_file(write_file, jointext(config_lines[filename], "\n\n"))
+
+
+/datum/controller/subsystem/configuration/proc/load_files()
+
+	// Load values from file into an assoc list.
+	var/list/loaded_values = list()
+	var/list/write_defaults = list()
+	for(var/filename in load_configuration_from)
+
+		if(!fexists(filename))
+			write_defaults += filename
+			continue
+
+		var/list/lines = file2list(filename)
+		for(var/line in lines)
+			line = trim(line)
+			if(!line || length(line) == 0 || copytext(line, 1, 2) == "#")
+				continue
+			var/pos = findtext(line, " ")
+			var/config_key
+			var/config_value
+			if(pos)
+				config_key   = copytext(line, 1, pos)
+				config_value = copytext(line, pos + 1)
+			else
+				config_key =   line
+				config_value = TRUE
+			if(config_key)
+				config_key = lowertext(trim(config_key))
+				if(config_key in loaded_values)
+					PRINT_STACK_TRACE("Duplicate config value loaded for key '[config_key]' from file '[filename]'.")
+				loaded_values[config_key] = config_value
+
+	// Write any defaults that aren't already populated.
+	if(length(write_defaults))
+		write_default_configuration(write_defaults)
+
+	// Set our config values on the decls.
+	var/list/config_to_refresh = list()
+	var/list/all_config_decls = decls_repository.get_decls_of_subtype(/decl/config)
+	for(var/config_type in all_config_decls)
+		var/decl/config/config_option = all_config_decls[config_type]
+		if(config_option.uid in loaded_values)
+			if(set_config_value(config_type, loaded_values[config_option.uid], defer_config_refresh = TRUE))
+				config_to_refresh += config_option
+			loaded_values -= config_option.uid
+
+	// Do a refresh now that all values are populated.
+	for(var/decl/config/config_option as anything in config_to_refresh)
+		config_option.update_post_value_set()
+
+/datum/controller/subsystem/configuration/proc/load_event(filename)
+	var/event_info = safe_file2text(filename, FALSE)
+	if(event_info)
+		global.custom_event_msg = event_info
+
+/datum/controller/subsystem/configuration/proc/load_sql()
+	var/list/lines = file2list(load_sql_from)
+	for(var/line in lines)
+		if(!line)
+			continue
+		line = trim(line)
+		if (length(line) == 0 || copytext(line, 1, 2) == "#")
+			continue
+
+		var/pos = findtext(line, " ")
+		var/name = null
+		var/value = null
+		if (pos)
+			name = lowertext(copytext(line, 1, pos))
+			value = copytext(line, pos + 1)
+		else
+			name = lowertext(line)
+		if (!name)
+			continue
+
+		switch (name)
+			if ("enabled")
+				sqlenabled = TRUE
+			if ("address")
+				sqladdress = value
+			if ("port")
+				sqlport = value
+			if ("database")
+				sqldb = value
+			if ("login")
+				sqllogin = value
+			if ("password")
+				sqlpass = value
+			else
+				log_misc("Unknown setting in configuration: '[name]'")
+
+/datum/admins/proc/dump_configuration()
```

Comment:
Seems like you got the same idea! Dumping options and writing the comments at the same time from code means the comments and value are more likely to match XD

Reply:
Alright. I liked the idea of having everything at the same place and just having a git action dump it or something. But, it's not like imperative tbh.